File: codeforces/1800~1899/1884/d.cc
#include <bits/stdc++.h>

using namespace std;

using i64 = long long;

void solve() {
  int n = 0;
  cin >> n;
  vector<int> a(n);
  for (int i = 0; i < n; ++i) cin >> a[i];

  int max_val = *max_element(a.begin(), a.end());
  vector<int> counts(max_val+1);
  for (int i = 0; i < n; ++i) {
    counts[a[i]]++;
  }

  vector<i64> dp(max_val+1);
  for (int g = max_val; g >= 1; --g) {
    int x = g;
    i64 total = 0;
    while (x <= max_val) {
      total += counts[x];
      x += g;
    }
    dp[g] = total * (total - 1) / 2;
  }
  for (int g = max_val; g >= 1; --g) {
    for (int j = g+g; j <= max_val; j += g) {
      dp[g] -= dp[j];
    }
  }

  vector<int> u = a;
  sort(u.begin(), u.end());
  u.erase(unique(u.begin(), u.end()), u.end());
  vector<bool> cannot_use(max_val+1);
  for (int val : u) {
    for (int i = val; i <= max_val; i += val) {
      cannot_use[i] = true;
    }
  }

  i64 ans = 0;
  for (int g = 1; g <= max_val; ++g) if (!cannot_use[g]) {
    ans += dp[g];
  }
  cout << ans << endl;
}

int main(void) {
  ios::sync_with_stdio(false);
  cin.tie(0);

  int T = 0;
  cin >> T;
  for (int t = 0; t < T; ++t) {
    solve();
  }
  return 0;
}

File: codeforces/1800~1899/1884/d.rs
use std::io::{stdin, stdout, BufWriter, Write};
use std::str::FromStr;
use std::fmt::Debug;

fn read_int<T: FromStr>() -> T where <T as FromStr>::Err: Debug {
    let mut s = String::new();
    stdin().read_line(&mut s).unwrap();
    s.trim().parse::<T>().expect("invalid input")
}

fn read_vec<T: FromStr>() -> Vec<T> where <T as FromStr>::Err: Debug {
    let mut s = String::new();
    stdin().read_line(&mut s).unwrap();
    s.split(" ").map(|x| x.trim().parse().expect("invalid input")).collect()
}

fn solve() {
    let stdout = stdout();
    let mut writer = BufWriter::new(stdout.lock());

    let n = read_int::<usize>();
    let a = read_vec::<i32>();
    let max_val = *a.iter().max().unwrap() as usize;
    let mut counts: Vec<i64> = vec![0; max_val+1];
    for i in 0..n {
        counts[a[i] as usize] += 1;
    }

    let mut dp = vec![0; max_val+1];
    for g in (1..=max_val).rev() {
        let mut total: i64 = 0;
        let mut x = g;
        while x <= max_val {
            total += counts[x];
            x += g;
        }
        dp[g] = total * (total - 1) / 2;
    }
    for g in (1..=max_val).rev() {
        for j in (g*2..=max_val).step_by(g) {
            dp[g] -= dp[j];
        }
    }

    let mut cannot_use: Vec<bool> = vec![false; max_val+1];
    for g in 1..=max_val {
        if counts[g] > 0 {
            for j in (g..=max_val).step_by(g) {
                cannot_use[j] = true;
            }
        }
    }

    let mut ans: i64 = 0;
    for g in 1..=max_val {
        if !cannot_use[g] {
            ans += dp[g];
        }
    }
    
    writeln!(writer, "{}", ans).ok();
}

fn main() {
    let n = read_int::<i32>();
    for _t in 0..n {
        solve();
    }
}
